fix scalar-tensor issue with CrossEntropyLoss label smoothing, symintify

bdhirsh opened the pull request:
Fixes the fake tensor (and symint) issues with https://github.com/pytorch/pytorch/issues/98571.

In particular, `at::detail::scalar_tensor_static()` seems like a footgun when used in code that needs to be traceable, so we may want to audit its usages more carefully.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98848
* #98031
* #98030
* #98960



cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @soumith @desertfire

bdhirsh committed:
fix scalar-tensor issue with CrossEntropyLoss label smoothing, symintify

[ghstack-poisoned]

ezyang commented on aten/src/ATen/TensorIndexing.h:
```diff
@@ -369,8 +369,7 @@ static inline Tensor scalarToTensor(
   if (self_device == at::kCPU) {
-    return at::detail::scalar_tensor_static(
-        v, options.dtype_opt()->toScalarType(), self_device);
+    return at::scalar_tensor(v, options.device(self_device));
```
seems like you've lost the dtype here

ezyang commented:
Everything else looks good, but you need to fix ci

bdhirsh committed:
Update on "fix scalar-tensor issue with CrossEntropyLoss label smoothing, symintify"

Fixes the fake tensor (and symint) issues with https://github.com/pytorch/pytorch/issues/98571.

In particular, `at::detail::scalar_tensor_static()` seems like a footgun when used in code that needs to be traceable, so we may want to audit its usages more carefully.




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

bdhirsh committed:
Update on "fix scalar-tensor issue with CrossEntropyLoss label smoothing, symintify"

Fixes the fake tensor (and symint) issues with https://github.com/pytorch/pytorch/issues/98571.

In particular, `at::detail::scalar_tensor_static()` seems like a footgun when used in code that needs to be traceable, so we may want to audit its usages more carefully.




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

bdhirsh committed:
Update on "fix scalar-tensor issue with CrossEntropyLoss label smoothing, symintify"

Fixes the fake tensor (and symint) issues with https://github.com/pytorch/pytorch/issues/98571.

In particular, `at::detail::scalar_tensor_static()` seems like a footgun when used in code that needs to be traceable, so we may want to audit its usages more carefully.




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

bdhirsh committed:
Update on "fix scalar-tensor issue with CrossEntropyLoss label smoothing, symintify"

Fixes the fake tensor (and symint) issues with https://github.com/pytorch/pytorch/issues/98571.

In particular, `at::detail::scalar_tensor_static()` seems like a footgun when used in code that needs to be traceable, so we may want to audit its usages more carefully.




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

bdhirsh committed:
Update on "fix scalar-tensor issue with CrossEntropyLoss label smoothing, symintify"

Fixes the fake tensor (and symint) issues with https://github.com/pytorch/pytorch/issues/98571.

In particular, `at::detail::scalar_tensor_static()` seems like a footgun when used in code that needs to be traceable, so we may want to audit its usages more carefully.




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]
